### ISSUE TITLE
A few clarifying questions

### ISSUE THREAD

**Issue by cwarny** [index-wallet/IW-cadCAD#1]
This isn't really an issue but I wasn't sure where else to ask my questions. I'm trying to understand the code to maybe see if I could have neural net players play the game with potentially more complex topologies or a larger economy (more goods traded) and see what comes out of it. But I'd like to understand what the current code does, so here are a bunch of questions:

1. Is an agent’s `wallet` variable the *amount* of each currency he carries?

2. Does the current code assume each agent only produces one good?

3. What is an agent’s “demand” for a neighbor? Is it the demand for the one good the neighbor produces? 

4. I’m confused by the `pricing_assessments` variable. Can you explain what it means? Is it the conversion rate of the various currencies into USD? Is it an agent’s internal valuation in USD of each cause? Is it the “vector price” of the one good the agent is selling? If it reflects the price of something, I find the term “assessment” confusing, as it seems to imply that the agent *choses* prices, but prices typically aren’t *chosen* by anyone, they are given by the market.

5. What is the meaning of `np.dot(customer.wallet, pricing_assessments[neighbor])` (in `get_best_vendors`)? My guess: `pricing_assessments[neighbor]` is the “vector price” of the one good the vendor is selling and so `np.dot(customer.wallet, pricing_assessments[neighbor])` is the value the customer would get by spending his entire wallet on that one vendor. But then this value is multiplied by `customer.demand[neighbor] / vendor.price`, which I take to mean the unit value to the customer of the one good the vendor is selling. And I don’t understand the meaning of the result. The code calls is `utils_scale`, but it’s unclear what the meaning is.

6. What is an agent’s `price` property? This seems to be a scalar. Is it the scalar price of the one good the agent is selling? It’s unclear to me if goods have a scalar price or a vector price. 

7. I don’t understand the profit function:
```
profit += (
    np.dot(customer.wallet, state["inherited_assessments"][node])
    / np.dot(customer.wallet, state["pricing_assessments"][node])
    / (1 + len(best_vendors[customer_idx]))
)
```
Can you explain how it works?
I also don't understand the optimization linked to the profit function.

8. I don't understand what "inherited assessment" means. What is the difference with pricing assessment?

**Comment by btgould** [comment]
> 1. Is an agent’s `wallet` variable the *amount* of each currency he carries?

Yes. The wallet of each agent is the current amount of each currency they possess.

> 2. Does the current code assume each agent only produces one good?

Yes. 
However, every agent's good is unique from that of any other agent. When people buy goods, we allow them to value some higher than others (currently assigned randomly).

> 3. What is an agent’s “demand” for a neighbor? Is it the demand for the one good the neighbor produces? 

Yes, this. Specifically, when agents are trying to decide which is their "best vendor", they look at the difference between their demand for that vendor's good and the loss in value in their wallet they incur by paying for the good. 

> 4. I’m confused by the `pricing_assessments` variable. Can you explain what it means? Is it the conversion rate of the various currencies into USD? Is it an agent’s internal valuation in USD of each cause? Is it the “vector price” of the one good the agent is selling? If it reflects the price of something, I find the term “assessment” confusing, as it seems to imply that the agent *choses* prices, but prices typically aren’t *chosen* by anyone, they are given by the market.

The code currently does not have any way to relate the value of any of the tokens to USD (caused by difficulties not related to this question, happy to explain more if desired). Each agent 

> 5. What is the meaning of `np.dot(customer.wallet, pricing_assessments[neighbor])` (in `get_best_vendors`)? My guess: `pricing_assessments[neighbor]` is the “vector price” of the one good the vendor is selling and so `np.dot(customer.wallet, pricing_assessments[neighbor])` is the value the customer would get by spending his entire wallet on that one vendor. But then this value is multiplied by `customer.demand[neighbor] / vendor.price`, which I take to mean the unit value to the customer of the one good the vendor is selling. And I don’t understand the meaning of the result. The code calls is `utils_scale`, but it’s unclear what the meaning is.
> 
> 6. What is an agent’s `price` property? This seems to be a scalar. Is it the scalar price of the one good the agent is selling? It’s unclear to me if goods have a scalar price or a vector price. 
> 
> 7. I don’t understand the profit function:
> ```
> profit += (
>     np.dot(customer.wallet, state["inherited_assessments"][node])
>     / np.dot(customer.wallet, state["pricing_assessments"][node])
>     / (1 + len(best_vendors[customer_idx]))
> )
> ```
> Can you explain how it works?
> I also don't understand the optimization linked to the profit function.
> 
> 8. I don't understand what "inherited assessment" means. What is the difference with pricing assessment?
